index.js

```diff
@@ -139,19 +141,24 @@ const pollWorkflow = () => {
     .catch((error) => {
       startGroup("Failed to Poll CircleCI Workflow");
       coreError(error);
-      setFailed(error.message);
+      retryCount++;
+      if (retryCount < maxRetries) {
+        info(`Retrying... (${retryCount}/${maxRetries})`);
+      } else {
+        setFailed(`Failed after ${maxRetries} retries: ${error.message}`);
+        followWorkflow = false;
+      }
       endGroup();
-      followWorkflow = false;
     });
 };
 
 if (followWorkflow) {
   info("Polling CircleCI Workflow");
 }
 
-const checkWebsiteStatus = setInterval(() => {
-  if (!followWorkflow) {
-    clearInterval(checkWebsiteStatus);
+const checkWorkflowStatus = setInterval(() => {
```

Comment:
I think there are some issues with this approach. It requires we `await` at the top level, which is difficult when we are writing test cases.

It would be good for us to use libraries like [axios-retry](https://github.com/softonic/axios-retry) to do this. It gives better control, and allows us to configure things like `retryCondition`, which would let us do it for our condition along with all network errors.